pkg/operator/baremetal_pod.go

```diff
@@ -234,13 +237,40 @@ func createMetal3PasswordSecrets(client coreclientv1.SecretsGetter, config *Oper
 	return nil
 }
 
+// Return false on error or if "baremetal.openshift.io/owned" annotation set
+func checkMetal3DeploymentMAOOwned(client appsclientv1.DeploymentsGetter, config *OperatorConfig) (bool, error) {
+	existing, err := client.Deployments(config.TargetNamespace).Get(context.Background(), baremetalDeploymentName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return true, nil
```

Comment:
i'm not understanding why this condition returns true. doesn't this mean that the baremetal deployment cannot be found and thus is not owned by anyone?

Reply:
Currently, in 4.6, MAO owns metal3 deployment by default. So, if MAO cannot find the deployment, it will go ahead and create one in syncBaremetalControllers().